File: 609-find-duplicate-file-in-system/609-find-duplicate-file-in-system.py
class Solution:
    def findDuplicate(self, paths: List[str]) -> List[List[str]]:
        contents_to_files = defaultdict(list)
        
        for path in paths:
            directory, *files = path.split()
            for file in files:
                # fname, contents = re.match(r"(.*\.txt)\((.*)\)", file).groups()
                fname, contents = file.split("(")
                contents_to_files[contents].append(f"{directory}/{fname}")
        
        return [fnames for fnames in contents_to_files.values() if len(fnames) > 1]

File: 609-find-duplicate-file-in-system/609-find-duplicate-file-in-system.js
/**
 * @param {string[]} paths
 * @return {string[][]}
 */
var findDuplicate = function(paths) {
    const contentToPath = {}
    
    paths.forEach((path) => {
        const [directory, ...files] = path.split(' ')
        
        files.forEach((file) => {
            const [fileName, contents] = file.split('(')
            const filePath = directory + '/' + fileName
            
            if (contentToPath[contents]) {
                contentToPath[contents].push(filePath)
            } else {
                contentToPath[contents] = [filePath]
            }
        })
    })
    
    return Object.values(contentToPath).filter((paths) => paths.length > 1)
};